flytectl/cmd/core/cmd.go

```diff
@@ -53,10 +53,18 @@ func generateCommandFunc(cmdEntry CommandEntry) func(cmd *cobra.Command, args []
 
 		if !cmdEntry.ProjectDomainNotRequired {
 			if config.GetConfig().Project == "" {
-				return fmt.Errorf("project and domain are required parameters")
+				if config.GetTaskConfig().Project != "" {
```

Comment:
We can merge the config and taskConfig [here](https://github.com/flyteorg/flyte/blob/ebad67cf2c29c1edab4a0137a4140f90db9be141/flytectl/cmd/config/config.go#L46-L48) instead